src/main/scala/com/github/unchama/seichiassist/subsystems/bookedachivement/bukkit/listener/GrantBookedAchievementListener.scala

```diff
@@ -0,0 +1,37 @@
+package com.github.unchama.seichiassist.subsystems.bookedachivement.bukkit.listener
+
+import cats.data.EitherT
+import cats.effect.IO
+import com.github.unchama.generic.effect.unsafe.EffectEnvironment
+import com.github.unchama.seichiassist.SeichiAssist
+import org.bukkit.event.player.PlayerJoinEvent
+import org.bukkit.event.{EventHandler, Listener}
+import com.github.unchama.targetedeffect.TargetedEffect
+import org.bukkit.command.CommandSender
+
+class GrantBookedAchievementListener(implicit effectEnvironment: EffectEnvironment) extends Listener {
+  @EventHandler
+  def onPlayerJoin(playerJoinEvent: PlayerJoinEvent): Unit = {
+    val player = playerJoinEvent.getPlayer
+    val uuid = player.getUniqueId
+    val playerData = SeichiAssist.playermap(uuid)
+
+    val manipulator = SeichiAssist.databaseGateway.bookedAchievementManipulator
+    import manipulator._
+
+    val effect = {
+      for {
+        bookedAchievementIds <- EitherT(loadNotGivenBookedAchievementsOf(player))
+        _ <- EitherT.right[TargetedEffect[CommandSender]](
+          IO {
+            bookedAchievementIds
+              .foreach(playerData.tryForcefullyUnlockAchievement)
+          }
+        )
```

Comment:
エフェクトを実行できてないです

`cats.implicits._` を import した上で、

```suggestion
        _ <- EitherT.right[TargetedEffect[CommandSender]](
          bookedAchievementIds
            .traverse(playerData.tryForcefullyUnlockAchievement)
        )
```

で通ったりしませんか？

Reply:
`.traverse` のところで `No implicit arguments of type: Applicative[TargetedEffect]` になってしまいます。

Reply:
そうか、 `Kleisli[F, *, Unit]` は確かにApplicativeではないですね…

```Scala
        _ <- EitherT.right[TargetedEffect[CommandSender]](
          SequentialEffect(
            bookedAchievementIds.map(playerData.tryForcefullyUnlockAchievement)
          )
        )
```

で良さそう？